http_requests.py

```diff
@@ -75,40 +77,70 @@ def mount_default_adapter(
     return session
 
 
+def _create_logging_client():
+    '''Create an ELS Client for the default els in the active config set and return it
+
+    Returns `None` if the config set has no default ELS config. Raises an error if the active
+    config set cannot be determined (`CONCOURSE_CURRENT_CFG` env var must be set)
+    '''
+    config_set_name = util.check_env('CONCOURSE_CURRENT_CFG')
+    config_set = ctx().cfg_factory().cfg_set(config_set_name)
+    try:
+        elastic_cfg = config_set.elasticsearch()
+    except KeyError:
+        return None
+    return ccc.elasticsearch.from_cfg(elasticsearch_cfg=elastic_cfg)
+
+
+def _log_entry_consumer(
+    queue,
+    elastic_client,
+    elastic_index_name,
+):
+    while True:
+        log_item = queue.get()
+        if(elastic_client):
+            try:
+                elastic_client.store_document(
+                    index=elastic_index_name,
+                    body=log_item,
+                )
+            except Exception as e:
+                info(f'Could not log stack trace information: {e}')
+
+
+if util._running_on_ci():
+    ELS_LOGGING_INDEX_NAME = 'github_access_stacktrace' # TODO: Make configurable
```

Comment:
still, this is not what I meant with my previous comment. also, why UPPERCASE?

please first create a separate commit / PR that factors out the creation / retrieval of elasticsearch client into a separate function, then update this PR so that it only introduces the asynchronous processing